test/script_tools.py

```diff
@@ -88,3 +89,60 @@ def test_threshold_custom_setting(self, mock_proceed, mock_lock, mock_lockfile):
         self.assertEqual(opts.nagios_reporter._cache_user, 'nrpe')
         self.assertTrue(opts.nagios_reporter._world_readable)
         self.assertTrue(opts.options.nagios_world_readable_check)
+
+
+magic = mock.MagicMock(name='magic')
+
+class MyCLI(CLI):
+    TIMESTAMP_MANDATORY = False  # mainly for testing, you really should need this in production
+    CLI_OPTIONS = {
+        'magic': ('some magic', None, 'store', 'magicdef'),
+    }
+    def do(self, dry_run):
+        return magic.go()
+
+
+class TestCLI(TestCase):
+    """Tests for the CLI base class"""
+
+    @mock.patch('vsc.utils.script_tools.ExtendedSimpleOption.prologue')
+    def test_opts(self, prol):
+        sys.argv = ['abc']
+        ms = MyCLI()
+
+        logging.debug("options %s %s %s", ms.options, dir(ms.options), vars(ms.options))
+
+        extsimpopts = {
+            'configfiles': None,
+            'debug': False,
+            'disable_locking': False,
+            'dry_run': False,
+            'ha': None,
+            'help': None,
+            'ignoreconfigfiles': None,
+            'info': False,
+            'locking_filename': '/var/lock/setup.lock',
+            'nagios_check_filename': '/var/cache/setup.nagios.json.gz',
+            'nagios_check_interval_threshold': 0,
```

Comment:
I never understood the purpose of this one.

Reply:
It's hazy for me as well, but it had something to do with rechecks from the monitoring?

Reply:
what would the script need to know about that?

Reply:
icinga itself schedules and reschedules according to its config. how would that affect the script? (and how is it kept in sync with what icinga schedules?)

Reply:
All right, but this PR is not the place to change this, I think.

Reply:
but if it is not needed, why define it?

Reply:
nagios.py has
~~~
       @param threshold: Seconds to determines how old the gzipped JSON data may be
                         before reporting an unknown result. This can be used to check if the script that uses the
                         reporter has run the last time and succeeded in writing the cache data. If the threshold <= 0,
                         this feature is not used.
~~~

Reply:
ach, so the name is just odd and confusing. (and only needed in corner cases.)